src/Components/Popover/Popover.tsx

```diff
@@ -0,0 +1,476 @@
+// Libraries
+import React, {Component, ReactChild, createRef, CSSProperties} from 'react'
+import classnames from 'classnames'
+
+// Components
+import {ClickOutside} from '../ClickOutside/ClickOutside'
+import {PopoverDismissButton} from './PopoverDismissButton'
+
+// Styles
+import './Popover.scss'
+
+// Types
+import {
+  ComponentColor,
+  StandardProps,
+  PopoverInteraction,
+  PopoverPosition,
+  PopoverType,
+} from '../../Types'
+
+interface Props extends StandardProps {
+  /** Popover dialog color */
+  color: ComponentColor
+  /** Popover dialog contents */
+  contents: (onHide?: () => void) => ReactChild
+  /** Type of interaction to show the popover dialog */
+  showEvent: PopoverInteraction
+  /** Type of interaction to hide the popover dialog */
+  hideEvent: PopoverInteraction
+  /** Pixel distance between trigger and popover dialog */
+  distanceFromTrigger: number
+  /** Where to position the popover relative to the trigger (assuming it fits there) */
+  position: PopoverPosition
+  /** Means of applying color to popover */
+  type: PopoverType
+  /** Renders the popover dialog visible initially */
+  initiallyVisible: boolean
+}
+
+interface State {
+  expanded: boolean
+  triggerRect: ClientRect | DOMRect | null
+}
+
+interface PopoverFlush {
+  first: boolean
+  last: boolean
+}
+
+export class Popover extends Component<Props, State> {
+  public static readonly displayName = 'Popover'
+
+  public static defaultProps = {
+    color: ComponentColor.Primary,
+    testID: 'popover',
+    showEvent: PopoverInteraction.Click,
```

Comment:
I know this isn't the time or place... but I think clockface would be a lot smoother to use if it presented types like

`type PopoverInteraction = 'click' | 'hover'`

instead of

```
enum PopoverInteraction {
  Click = 'click',
  Hover = 'hover',
}
```